apiary.apib

```diff
@@ -0,0 +1,152 @@
+CSHARPDEFAULTNAMESPACE: DSI.SDK.Settings
+JAVADEFAULTNAMESPACE: io.dudesolutions.settings
+BRANDLABEL: Dude Solutions, Inc.
+USERAGENT: Dude SDK
+
+# Dude Settings Api
+![Dude Solutions](https://dudesolutions.com/assets/img/logo.png)
+
+## Authentication
+*Dude Api* uses OAuth v2 Authorization. First you create a new (or acquire existing) OAuth v2 token using [DudeAuth](https://www.dudeauth.com/).
+
+New to *OAuth*?  Check out the [OAuth Bible](http://oauthbible.com/) for good reference material.
+
+## Media Types
+Where applicable this API uses the *JSON* media-type to represent resources.
+
+Requests and responses with a message-body use plain JSON to set or update resources.
+
+## Error States
+Common [HTTP Response Status Codes](https://github.com/for-GET/know-your-http-well/blob/master/status-codes.md) are used throughout this API.
+
+In addition to standard response codes the response object returned from the server during an exception event may contain additional information in the following JSON format:
+
+        {
+            "message": "SOME MESSAGE",
+            "dudeReferenceId": "3523845A-4729-4CCB-9E16-01DEADF6C2DF",
+            "stackTrace": "Null, Typically"
+        }
+
+## Data Restriction
+Depending on the access level granted to the calling client certain data elements throughout this API will be removed and not returned.
+
+## Dates & Times
+All dates and times are represented in [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) format.  In addition, *Dude Api* will always provide [Coordinated Universal Time](https://en.wikipedia.org/wiki/Coordinated_Universal_Time).
+
+The API is capable of parsing most date and time formats, however we do require UTC dates and times be provided from all calling systems.  We highly recommend you utilize the ISO-8601 standard when communicating dates and time to the API as well.
+
+## Known Issues
+None at this time
+
+## Documentation Notes
+In Progress
+
+# Group System Information
+
+## System Information [/]
+
+### GET
+
++ Response 200 (application/json)
+    + Attributes (System Information)
+
+# Group Health
+
+## Heartbeat [/heartbeat]
+The heartbeat is exposed and provided to check for availability of the service.
+Any response that does not provide a ```200 OK``` response code should be treated as an error.
+
+### GET
+
++ Response 200 (application/json)
+    + Attributes (Heartbeat)
+
+# Group Account Settings
+
+## Account Setting Collection [/accounts/{accountId}/settings{?page,pagesize,orderby}]
+
++ Parameters
+    + accountId: 1 (required, number) - {INT} ID of the account in the form of an integer
+    + page: 1 (number, optional) - {INT} The page to request from the collection.
+    + pagesize: 1000 (number, optional) - {INT} The number of records to load in a page from the collection.
+
+### GET
+
++ Response 200 (application/json)
+    + Attributes (Base Collection Response)
+        + `data` (array[Account Setting], required)
+
+### POST
```

Comment:
We have been creating a separate collection for POST so the endpoint model doesn't include the query parameters.  If we want to change that now that we have the parameters included in a endpoint model we should update the documentation here: https://readme.dudesoln.com/display/EA/Apiary+Documentation+Standards